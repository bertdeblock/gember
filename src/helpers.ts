import type { EmberPackageJson } from "./types.js";

export function isV1Addon(packageJson: EmberPackageJson): boolean {
  if (isAddon(packageJson)) {
    const { version } = packageJson["ember-addon"] ?? {};

    return version === 1 || version === undefined;
  }

  return false;
}

export function isV2Addon(packageJson: EmberPackageJson): boolean {
  if (isAddon(packageJson)) {
    const { version } = packageJson["ember-addon"] ?? {};

    return version === 2;
  }

  return false;
}

function isAddon(packageJson: EmberPackageJson): boolean {
  if (Array.isArray(packageJson.keywords)) {
    return packageJson.keywords.includes("ember-addon");
  }

  return false;
}
