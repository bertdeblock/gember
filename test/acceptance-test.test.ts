import { afterEach, it } from "vitest";
import { Package } from "./helpers.ts";

let pkg: Package;

afterEach(() => pkg.cleanUp());

it("generates a `.js` acceptance-test", async (ctx) => {
  pkg = await Package.create("v2-addon");

  await pkg.gember("acceptance-test", "foo");

  const content = await pkg.readFile("tests/acceptance/foo-test.js");

  ctx.expect(content).toMatchSnapshot();
});

it("generates a `.js` acceptance-test at a custom path", async (ctx) => {
  pkg = await Package.create("v2-addon");

  await pkg.gember("acceptance-test", "foo", "--path=tests/foo");

  const content = await pkg.readFile("tests/foo/foo-test.js");

  ctx.expect(content).toMatchSnapshot();
});

it("generates a `.ts` acceptance-test", async (ctx) => {
  pkg = await Package.create("v2-addon");

  await pkg.gember("acceptance-test", "foo", "--ts");

  const content = await pkg.readFile("tests/acceptance/foo-test.ts");

  ctx.expect(content).toMatchSnapshot();
});

it("generates a `.ts` acceptance-test at a custom path", async (ctx) => {
  pkg = await Package.create("v2-addon");

  await pkg.gember("acceptance-test", "foo", "--path=tests/foo", "--ts");

  const content = await pkg.readFile("tests/foo/foo-test.ts");

  ctx.expect(content).toMatchSnapshot();
});
