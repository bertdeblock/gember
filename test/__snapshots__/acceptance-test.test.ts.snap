// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generates a \`.js\` acceptance-test 1`] = `
"import { currentURL, visit } from '@ember/test-helpers';
import { module, test } from 'qunit';
import { setupApplicationTest } from 'v2-addon/tests/helpers';

module('Acceptance | foo', function (hooks) {
  setupApplicationTest(hooks);

  test('it visits /foo', async function (assert) {
    await visit('/foo');

    assert.strictEqual(currentURL(), '/foo');
  });
});
"
`;

exports[`generates a \`.js\` acceptance-test at a custom path 1`] = `
"import { currentURL, visit } from '@ember/test-helpers';
import { module, test } from 'qunit';
import { setupApplicationTest } from 'v2-addon/tests/helpers';

module('Acceptance | foo', function (hooks) {
  setupApplicationTest(hooks);

  test('it visits /foo', async function (assert) {
    await visit('/foo');

    assert.strictEqual(currentURL(), '/foo');
  });
});
"
`;

exports[`generates a \`.ts\` acceptance-test 1`] = `
"import { currentURL, visit } from '@ember/test-helpers';
import { module, test } from 'qunit';
import { setupApplicationTest } from 'v2-addon/tests/helpers';

module('Acceptance | foo', function (hooks) {
  setupApplicationTest(hooks);

  test('it visits /foo', async function (assert) {
    await visit('/foo');

    assert.strictEqual(currentURL(), '/foo');
  });
});
"
`;

exports[`generates a \`.ts\` acceptance-test at a custom path 1`] = `
"import { currentURL, visit } from '@ember/test-helpers';
import { module, test } from 'qunit';
import { setupApplicationTest } from 'v2-addon/tests/helpers';

module('Acceptance | foo', function (hooks) {
  setupApplicationTest(hooks);

  test('it visits /foo', async function (assert) {
    await visit('/foo');

    assert.strictEqual(currentURL(), '/foo');
  });
});
"
`;
